server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterGetSettingsAction.java

```diff
@@ -50,16 +60,46 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        ClusterStateRequest clusterStateRequest = Requests.clusterStateRequest().routingTable(false).nodes(false);
         final boolean renderDefaults = request.paramAsBoolean("include_defaults", false);
+
+        if (nodesInCluster.get().getMinNodeVersion().before(Version.V_8_3_0)) {
+            return prepareLegacyRequest(request, client, renderDefaults);
+        }
+
+        ClusterGetSettingsAction.Request clusterSettingsRequest = new ClusterGetSettingsAction.Request();
+
+        clusterSettingsRequest.local(request.paramAsBoolean("local", clusterSettingsRequest.local()));
+        clusterSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterSettingsRequest.masterNodeTimeout()));
+
+        return channel -> client.execute(
+            ClusterGetSettingsAction.INSTANCE,
+            clusterSettingsRequest,
+            new RestToXContentListener<RestClusterGetSettingsResponse>(channel).map(
+                r -> response(r, renderDefaults, settingsFilter, clusterSettings, settings)
```

Comment:
I'd still sort of like to reduce the duplication here between the two branches but I see it's not so simple because we have to consume the request params before we get hold of the channel. Not a blocking suggestion.